### PR TITLE
Update to use MitoAnalysis class

### PR DESCRIPTION
**Pull request by marthacryan**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -102,51 +97,33 @@ def get_code_to_write(description, code):
...
-
-        # Allow users to download this new dataframe
-
-        st.success("Finished automation.")
+        analysis = MitoAnalysis.from_json(f.read())
+
+    # Read in params
+    updated_metadata = {}
+    for param in analysis.get_param_metadata(param_type='import'):
+        new_param = st.file_uploader(param['name'])
```
Comment (naterush) [inline main.py]: Think you want to make this CSVs only

File: main.py [hunk 2]
```diff
@@ -102,51 +97,33 @@ def get_code_to_write(description, code):
...
+
+    run = st.button('Run')
+    if run:
+        print('running')
+        result = analysis.run(**updated_metadata)
 
         @st.cache_data
         def convert_df(df):
             # IMPORTANT: Cache the conversion to prevent computation on every rerun
             return df.to_csv().encode('utf-8')
```
Comment (naterush) [inline main.py:118]: This line of code is erroring. See error here:
Screen Shot 2023-10-19 at 9 03 39 PM

**Review by naterush — CHANGES_REQUESTED** [review]
Some bugs to fix up here, I think.

File: main.py (hunk 2 above)
Comment (marthacryan) [reply main.py:118]: Oops fixed. This was because I was assuming the result was a list but it was actually sometimes just a DF.

File: main.py [hunk 3]
```diff
@@ -102,51 +97,35 @@ def get_code_to_write(description, code):
...
+        downloads = []
+        if isinstance(result, pd.DataFrame):
+            downloads.append(convert_df(result))
+        elif isinstance(result, tuple) and len(result) > 0:
+            for res in result:
+                downloads.append(convert_df(res))
         st.download_button(
             label="Download final automation as CSV",
-            data=csv,
+            data=downloads[0],
```
Comment (naterush) [inline main.py]: I think we want the -1 file, rather than the first file.

**Review by naterush — CHANGES_REQUESTED** [review]
One minor change, then LGTM

**Pull request merged**